source/elements/oneDAL/source/terminology.rst

```diff
@@ -0,0 +1,64 @@
+.. _terminology:
+
+============
+Terminology
+============
+
+@table-begin:
+   column-sizes: [ 25% ]
+Term | Definition \\
+
+Batch Mode | The computation mode for an algorithm in oneDAL, where all the data needed for
+computation are available at the start and fits the memory of the device where computations will
+be performed. \\
+
+Dataset | A pair of a matrix $X$ of size $N \times p$ and a vector $Y$ of size $N$, where $X$
+contains $N$ feature vectors in rows and $Y$ stores all corresponding responses. Vector $Y$ is
```

Comment:
We refer to 'feature vector' term which is not defined - it is defined indeed, however is not in the separate line - need to fix.  It makes sense to set-up a closer connection to the 'observation' in this def and other related terms below to highlight their dependence on each other

Reply:
Per offline discussion with @outoftardis, we can use the style of defs shown at https://outoftardis.github.io/daal/dev_guide/algorithms/dbscan/index.html as the working option for the whole spec

Reply:
Switch terminology section to proposed format and added cross reference between terms
![image](https://user-images.githubusercontent.com/18015221/77199867-95740b80-6afa-11ea-8dfe-73153241d62e.png)